### PR TITLE
Expose promise interface in auth module

### PR DESCRIPTION
**Pull request by jviotti**
This introduces a small breaking change:

- auth.authenticate() doesn't yield a username anymore.

### CODE REVIEW

File: lib/auth.coffee [hunk 1]
```diff
@@ -64,20 +59,19 @@ exports.whoami = (callback) ->
...
-	request.request
-		method: 'POST'
-		url: '/login_'
-		json: credentials
-	, (error, response) ->
-		return callback(error) if error?
-		savedToken = response?.body
-		return callback(null, savedToken, credentials.username)
+	Promise.fromNode (callback) ->
+		request.request
```
Comment (emirotin) [inline lib/auth.coffee]: I think you can `promisifyAll` request instead

Comment (Page-) [inline lib/auth.coffee]: You can, and then `request.postAsync().spread (response, body) ->`

Comment (jviotti) [inline lib/auth.coffee]: Sadly  `.spread' did not work on this function:

```sh
TypeError: expecting an array, a promise or a thenable. See http://goo.gl/s8MMhc
```

So I had to use `.then` here, but it worked in `.register()`.

File: lib/auth.coffee [hunk 2]
```diff
@@ -269,20 +258,10 @@ exports.logout = (callback) ->
...
-
-	request.request
+	request.requestAsync
 		method: 'POST'
 		url: '/user/register'
 		json: credentials
-	, (error, response, body) ->
-		return callback(error) if error?
-		return callback(null, body)
+	.spread (response, body) ->
```
Comment (Page-) [inline lib/auth.coffee]: This can be `.get(1)`

File: lib/auth.coffee [hunk 3]
```diff
@@ -64,20 +59,18 @@ exports.whoami = (callback) ->
...
-	request.request
+	request.requestAsync
 		method: 'POST'
 		url: '/login_'
 		json: credentials
-	, (error, response) ->
-		return callback(error) if error?
-		savedToken = response?.body
-		return callback(null, savedToken, credentials.username)
+	.then (data) ->
```
Comment (Page-) [inline lib/auth.coffee]: This can be `.get('body')`

File: lib/auth.coffee (hunk 1 above)
Comment (Page-) [inline lib/auth.coffee]: That is very weird, since they are both POSTs :S

File: lib/auth.coffee (hunk 2 above)
Comment (jviotti) [inline lib/auth.coffee]: Fixed.

File: lib/auth.coffee (hunk 3 above)
Comment (jviotti) [inline lib/auth.coffee]: Fixed.

**Comment by Page-** [comment]
LGTM

**Pull request merged**